react-components/src/architecture/concrete/primitives/cylinder/CylinderView.ts

```diff
@@ -307,9 +338,99 @@ export class CylinderView extends GroupThreeView<CylinderDomainObject> {
   }
 
   // ==================================================
-  // INSTANCE METHODS: For picking
+  // INSTANCE METHODS: Labels
   // ==================================================
 
+  private addLabels(): void {
+    this._radiusSprite = undefined;
+    this._radiusSprite = undefined;
+    const type = this.domainObject.primitiveType;
+    if (type === PrimitiveType.VerticalCylinder || type === PrimitiveType.HorizontalCircle) {
+      this._radiusSprite = this.createRadiusLabel();
+      if (this._radiusSprite !== undefined) {
+        this.addChild(this._radiusSprite);
+      }
+    }
+    if (type === PrimitiveType.VerticalCylinder) {
+      this._heightSprite = this.createHeightLabel();
+      if (this._heightSprite !== undefined) {
+        this.addChild(this._heightSprite);
+      }
+    }
+  }
+
+  private getRadiusTextHeight(relativeTextSize: number): number {
+    return relativeTextSize * 2 * this.domainObject.cylinder.radius;
+  }
+
+  private getHeightTextHeight(relativeTextSize: number): number {
+    return relativeTextSize * this.domainObject.cylinder.height;
+  }
+
+  private createRadiusLabel(): Sprite | undefined {
+    const { style, domainObject } = this;
+    const radius = domainObject.cylinder.radius;
+    if (!Cylinder.isValidSize(radius)) {
+      return undefined; // Not show when about 0
+    }
+    const spriteHeight = this.getRadiusTextHeight(style.relativeTextSize);
+    const unitSystem = this.getUnitSystem();
+    const text = unitSystem.toStringWithUnit(radius, Quantity.Length);
+    return createSprite(text, this.style, spriteHeight);
+  }
+
+  private createHeightLabel(): Sprite | undefined {
+    const { domainObject, style } = this;
+    const height = domainObject.cylinder.height;
+    if (!Cylinder.isValidSize(height * 0.99)) {
+      return undefined; // Not show when about 0
+    }
+    const spriteHeight = this.getHeightTextHeight(style.relativeTextSize);
+    const unitSystem = this.getUnitSystem();
+    const text = unitSystem.toStringWithUnit(height, Quantity.Length);
+    return createSprite(text, this.style, spriteHeight);
+  }
+
```

Comment:
These are almost identical. I think a little helper adjacent to them would be worth it.